src/daft-local-execution/src/intermediate_ops/aggregate.rs

```diff
@@ -0,0 +1,34 @@
+use std::sync::Arc;
+
+use common_error::DaftResult;
+use daft_dsl::ExprRef;
+use daft_micropartition::MicroPartition;
+
+use super::intermediate_op::IntermediateOperator;
+
+#[derive(Clone)]
+pub struct AggregateOperator {
+    agg_exprs: Vec<ExprRef>,
+    group_by: Vec<ExprRef>,
+}
+
+impl AggregateOperator {
+    pub fn new(agg_exprs: Vec<ExprRef>, group_by: Vec<ExprRef>) -> Self {
+        Self {
+            agg_exprs,
+            group_by,
+        }
+    }
+}
+
+impl IntermediateOperator for AggregateOperator {
+    fn execute(&self, input: &Arc<MicroPartition>) -> DaftResult<Arc<MicroPartition>> {
+        log::debug!("AggregateOperator::execute");
+        let out = input.agg(&self.agg_exprs, &self.group_by)?;
+        Ok(Arc::new(out))
+    }
+
+    fn name(&self) -> String {
```

Comment:
no need to block here but this trait should return a `&'static str` so we don't need to perform a heap allocation each time we ask for the name